### ISSUE TITLE
GSheets is installing older extension version / how to install specific extension version?

### ISSUE THREAD

**Issue by Alex-Monahan** [duckdb/community-extensions#241]
Howdy folks and happy holidays!

From what I can see in the repo, the GSheets extension has been updated to the correct sha for version 0.0.3 of the extension. However, when I install the extension, I receive 0.0.2. 

Is there a way to get 0.0.3 to install by default?

In the meantime, is there a way to force the installation of a specific version?

Many thanks!

Here are the GitHub actions that should have built version 0.0.3:
* https://github.com/duckdb/community-extensions/actions/runs/11940610061
* https://github.com/duckdb/community-extensions/actions/runs/11939341143

I run into this issue on MacOS with both the CLI and Python using DuckDB version 1.1.3. I was expecting the extension version to be `v0.0.3`, but the query below shows `v0.0.2`. I was also seeing a bug re-appear that was fixed in v0.0.3, so I believe that the older version is what is getting installed.

```sql
D install gsheets from community;
D load gsheets;
D from duckdb_extensions() where extension_name = 'gsheets';
┌────────────────┬─────────┬───────────┬───┬───────────┬───────────────────┬──────────────┬────────────────┐
│ extension_name │ loaded  │ installed │ … │  aliases  │ extension_version │ install_mode │ installed_from │
│    varchar     │ boolean │  boolean  │   │ varchar[] │      varchar      │   varchar    │    varchar     │
├────────────────┼─────────┼───────────┼───┼───────────┼───────────────────┼──────────────┼────────────────┤
│ gsheets        │ true    │ true      │ … │ []        │ v0.0.2            │ REPOSITORY   │ community      │
├────────────────┴─────────┴───────────┴───┴───────────┴───────────────────┴──────────────┴────────────────┤
│ 1 rows                                                                               9 columns (7 shown) │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

FYI @archiewood

**Comment by carlopi** [comment]
Once an extension is locally installed, `INSTALL x FROM y` only checks that extension is actually there, but no attempt at updating are currently performed.

There are 2 main ways to bump to the most recently available version of an extension:
* `FORCE INSTALL x FROM y`: this will disregard any currently installed extension, and just redownload it
* `UPDATE EXTENSIONS (x)` or the general `UPDATE EXTENSIONS`: those will perform a GET request with a Etag, so that only on changes a new extension is downloaded.

**Comment by archiewood** [comment]
@Alex-Monahan i assume you have tried UPDATE EXTENSIONS;

https://duckdb.org/docs/extensions/overview.html#updating-extensions

update: @carlopi was faster

**Comment by carlopi** [comment]
Actually checked, `UPDATE EXTENSIONS` still gives `v0.0.2`, I am not actually sure, I might have to double check the binaries.

**Comment by Alex-Monahan** [comment]
Thanks folks! The update command will be super handy for me in the future. I agree that something is still up with the binaries though since I still see 0.0.2 as the latest. 

As a note, this is not urgent and can wait until after the holidays!!

```sql
D update extensions(gsheets);
┌────────────────┬────────────┬─────────────────────┬──────────────────┬─────────────────┐
│ extension_name │ repository │    update_result    │ previous_version │ current_version │
│    varchar     │  varchar   │       varchar       │     varchar      │     varchar     │
├────────────────┼────────────┼─────────────────────┼──────────────────┼─────────────────┤
│ gsheets        │ community  │ NO_UPDATE_AVAILABLE │ v0.0.2           │ v0.0.2          │
└────────────────┴────────────┴─────────────────────┴──────────────────┴─────────────────┘
```

**Comment by archiewood** [comment]
I have replicated this also

**Comment by archiewood** [comment]
@Alex-Monahan I just merged in v0.0.4 - let me know if this resolves for you

**Comment by Alex-Monahan** [comment]
This is resolved, yes! Thank you!
